subgraphs/rocket-pool/src/entityUpdates/financialMetrics.ts

```diff
@@ -0,0 +1,440 @@
+    getOrCreateToken(Address.fromString(ETH_ADDRESS), block.number)
+      .lastPriceUSD!
+  );
+
+  let rplTVLUSD = bigIntToBigDecimal(inputTokenBalances[0]).times(
+    getOrCreateToken(Address.fromString(RPL_ADDRESS), block.number)
+      .lastPriceUSD!
+  );
+
+  let totalValueLockedUSD = ethTVLUSD.plus(rplTVLUSD);
+  pool.totalValueLockedUSD = totalValueLockedUSD;
+  pool.save();
+
+  // Pool Daily and Hourly
+  // updateSnapshotsTvl(event.block) is called separately when protocol and supply side revenue
+  // metrics are being calculated to consolidate respective revenue metrics into same snapshots
+
+  // Protocol
+  protocol.totalValueLockedUSD = pool.totalValueLockedUSD;
+  protocol.save();
+
+  // Financials Daily
+  // updateSnapshotsTvl(event.block) is called separately when protocol and supply side revenue
+  // metrics are being calculated to consolidate respective revenue metrics into same snapshots
+}
+
+export function updateSnapshotsTvl(block: ethereum.Block): void {
+  const pool = getOrCreatePool(block.number, block.timestamp);
+  const poolMetricsDailySnapshot = getOrCreatePoolsDailySnapshot(
+    Address.fromString(PROTOCOL_ID),
+    block
+  );
+  const poolMetricsHourlySnapshot = getOrCreatePoolsHourlySnapshot(
+    Address.fromString(PROTOCOL_ID),
+    block
+  );
+  const financialMetrics = getOrCreateFinancialDailyMetrics(block);
+
+  // Pool Daily
+  poolMetricsDailySnapshot.totalValueLockedUSD = pool.totalValueLockedUSD;
+  poolMetricsDailySnapshot.inputTokenBalances = pool.inputTokenBalances;
+  poolMetricsDailySnapshot.save();
+
+  // Pool Hourly
+  poolMetricsHourlySnapshot.totalValueLockedUSD = pool.totalValueLockedUSD;
+  poolMetricsHourlySnapshot.inputTokenBalances = pool.inputTokenBalances;
+  poolMetricsHourlySnapshot.save();
+
+  // Financials Daily
+  financialMetrics.totalValueLockedUSD = pool.totalValueLockedUSD;
+  financialMetrics.save();
+}
+
+export function updateTotalRevenueMetrics(
+  block: ethereum.Block,
+  stakingRewards: BigDecimal,
+  totalShares: BigInt // of rETH
+): void {
+  const pool = getOrCreatePool(block.number, block.timestamp);
+  const protocol = getOrCreateProtocol();
+  const financialMetrics = getOrCreateFinancialDailyMetrics(block);
+  const poolMetricsDailySnapshot = getOrCreatePoolsDailySnapshot(
+    Address.fromString(PROTOCOL_ID),
+    block
+  );
+  const poolMetricsHourlySnapshot = getOrCreatePoolsHourlySnapshot(
+    Address.fromString(PROTOCOL_ID),
+    block
+  );
+
+  // Staking Rewards
+
+  // Pool
+  pool.cumulativeTotalRevenueUSD = stakingRewards;
+  pool.outputTokenSupply = totalShares;
+  pool.outputTokenPriceUSD = getOrCreateToken(
+    Address.fromString(PROTOCOL_ID),
+    block.number
+  ).lastPriceUSD;
+  pool.save();
+
+  // Pool Daily
+  poolMetricsDailySnapshot.cumulativeTotalRevenueUSD =
+    pool.cumulativeTotalRevenueUSD;
+  poolMetricsDailySnapshot.dailyTotalRevenueUSD = stakingRewards;
```

Comment:
`dailyTotalRevenueUSD` and `cumulativeTotalRevenueUSD` have the same value?

Reply:
adjusted methodology significantly to improve correctness.